### ISSUE TITLE
Failed to config.json with every update, and stays non responsive until delete old config...

### ISSUE THREAD

**Issue by nikhil-swamix** [continuedev/continue#2295]
### Before submitting your bug report

- [X] I believe this is a bug. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue) that reports the same bug
- [X] I've seen the [troubleshooting guide](https://docs.continue.dev/troubleshooting) on the Continue Docs

### Relevant environment info

```Markdown
its neglect to handle errors with config json. the system is curroupt.
```

### Description

![image](https://github.com/user-attachments/assets/e6c90fec-c3e9-4952-a4e8-3a770f853c52)

# hi, whenever an update is detected by continue, the config system starts throwing this message, ive seen this  happen 4 times, i mhave to make backup of conf file, start and wait for extension to create defaults, and replace with backup. this is ultimate inconvenience, where the extension was supposed to save time, is now requires hot fixes just to get it working... i appreciate the efforts being put, but without some basic checks, the fun reduced...

### To reproduce

Random, but noticed update extension in store whenever this is getting toasted...

### Log output

```Shell
}

Error: No handler for message type "config/getSerializedProfileInfo"
2024-09-18 15:56:13.594 [error] [Extension Host] Error handling webview message: {
  "msg": {
    "messageId": "9fb42620-9fdc-4300-a076-fdd3f04145c4",
    "messageType": "config/getSerializedProfileInfo"
  }
}

Error: No handler for message type "config/getSerializedProfileInfo"
2024-09-18 15:56:15.603 [error] [Extension Host] Error handling webview message: {
  "msg": {
    "messageId": "982907aa-8465-4dfa-a63b-2106eddbb9db",
    "messageType": "config/getSerializedProfileInfo"
  }
}

Error: No handler for message type "config/getSerializedProfileInfo"
2024-09-18 15:56:17.595 [error] [Extension Host] Error handling webview message: {
  "msg": {
    "messageId": "f777e0e0-0f09-44de-885e-913961cadc2c",
    "messageType": "config/getSerializedProfileInfo"
  }
}

Error: No handler for message type "config/getSerializedProfileInfo"
```

**Comment by nikhil-swamix** [comment]
no,, i delete the indexes, migrations, and diffs, except the config, in ~user folder, it autoupdate from store every now and then. thats the most reliable solution. i can 200% assure you it its independent of conf file, rather im more suspicious about, how I/O are handled on crashes and version updates. and its nearly impossible to reproduce, i tried doing... maybe some error handling raises, then file is kept open and corroupt... guessing..

**Comment by sestinj** [comment]
@nikhil-swamix Thanks for the extra info.

Even though it looks like not many users are experiencing this, it's critical for those who do, so I've made an attempt to catch errors from any possible `JSON.parse` call in our code that could be causing this. This new release is `v0.9.222`. Please let me know whether this addresses the problem.

But I'd still like to be able to understand the exact cause and there's unfortunately not quite enough information to reproduce this, as hard as I have tried. So far I have attempted:
- Creating an invalid config.json file
- Uninstalling/re-installing
- Downloading different recent versions, both downgrading and upgrading
- Repeating all of the above using a Remote SSH workspace

Can you share more information about your environment?
- OS/arch
- VS Code version
- Continue version
- Local/Remote SSH/Dev Containers/etc
- config.json when the error occurs

cc'ing @sdudzin @scottopell as well in case you can share anything else about your setup.

**Comment by nikhil-swamix** [comment]
appreciate your efforts on this issue, even i had difficulty on replication. my biggest doubt is on the sqlite database or index db, and the fact, my D drive was failing randomly! its a nvme m2 ssd, and my github lies there. i have updated the BIOS firmware for the sake of this issue! afaik, no issue since last week, im monitoring behaviour and will report when failed. not sure what other users may facing error. 

Regarding Env:
Windows 10 , 
vscode 1.94.2
continue 0.8.53 - 2024-10-11
no containers,
Config file is stable as hell, i just keep a backup between versions, using rollback command
```
$src = "C:\Users\User\.continue\config.json"
$dst = "C:\Users\User\.continue\config_backup_$(Get-Date -Format 'yyyyMMdd_HHmmss').json"
Copy-Item $src $dst
```
_____
maybe if you can add full traceback in inspector... i can dump here. the message in debug is unclear, and some random build hash was returned which made difficult to go deeper... 
1. maybe if source map can added in final build? 
2. maybe a continue.debug.getEnvDetailsForGithubIssue() like fuinction implemented, so user may post directly, and it will copy to clipboard?

you know better you may implement whats right. regards

**Comment by nikhil-swamix** [comment]
@sestinj 
![image](https://github.com/user-attachments/assets/4ba7b402-9683-4c37-86dd-084ddf6a59fe)

i updated version, not greeted with error, so this point "between uersion updates" may ruled out, a possibility is that, when i use ollama, continue starts to panic, i have "disable indexing on startup" setting for it. sometimes the usage of ollama and bge-m3 model is so severe that, i get Blue screen of death, and Kernel Inpage error... there is a chance that index DB may corroupt... cpu usage bumps 100%.

what i propose, is to limit the chunking being sent when indexing, via a prrameter. and allow it slowly. maybe a dropdown could added for most prefillable options, this setting seems a mega UX improvement
![image](https://github.com/user-attachments/assets/062fb25d-c29f-41fc-a76e-59d98b50843f)

hope you capitalize on this, 100 points awarded... 👍

**Comment by codeandcountry** [comment]
Seeing this error too
![image](https://github.com/user-attachments/assets/bd9be25a-08e5-4af0-8431-eb2e04a903cf)

**Comment by nikhil-swamix** [comment]
ok, the happiness was shortlived, i faced once 2 days ago. yet again, deleting all except (sessions,conf.json ) and restart vscode did the job. the devil is in the database. random power cuts while its responding, will kill the extension.
